File: isPrime.py
def isPrime(number):
    for n in range(2, number):
        if number % n == 0:
            return False
    return True

def isPrimeByFactors(number):
    factors = []
    for i in range(2, number):
        if number % i == 0: # i is a factor
            factors.append(i)
    if len(factors) == 0:
        return True
    else:
        print(factors)
        return False
        


##### Driver ######

def main():
    number = int(input("number to test: "))
    print(str(isPrimeByFactors(number)))

main()

File: isPrime.hs
factors n = [x | x <- [2..n - 1], n `mod` x == 0]
isPrime n = length (factors n) == 0

myPrime n = foldl (\acc x -> if n `mod` x == 0 then (False, x) else acc) (True, 0) [2..n - 1]
